basesandbox/src/ipc/mod.rs

```diff
@@ -0,0 +1,92 @@
+// Copyright 2020 Kodebox, Inc.
+// This file is part of CodeChain.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pub mod domain_socket;
+pub mod intra;
+pub mod semaphore;
+use std::collections::hash_map::HashMap;
+
+/// Inter-process obejct that is constructed / synchronized by another Inter-process mechanism.
+pub trait InterProcessUnit {
+    /// Constructs itself with an opaque data that would have been transported by some IPC
+    fn new(data: Vec<u8>) -> Self;
+}
+
+/// One possible abstraction for a server-client style of `Linker`
+pub trait TwoWayInitialize {
+    type Server: InterProcessUnit;
+    type Client: InterProcessUnit;
```

Comment:
I can't find actual uses of `Server` and `Client`.

Reply:
They are reserved for even higher abstraction that takes `T: TwoWayInitialize`, and perform generic way of utilizing it, possibly in the coordinator. Personally I enjoy this kind of further-use abstraction, but also agree that it has no reason to exit for now. I'll remove it after I check they are not actually used in FML part.